### PR TITLE
added messages for processing phases

### LINKED ISSUE #128: When you upload/download a large file, there is a noticeable delay until the progress bar is displayed
[Affected versions]:
- Firefox 54.0 and up

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Steps to reproduce]:
1. Start the Firefox browser with and navigate to "https://send.dev.mozaws.net" page.
2. Upload a large file (eg: 1.0 Gb) and observe the behavior.

[Expected result]:
- The progress bar is displayed even if the upload didn't start.
Or
- A loading spinner appears until the progress bar is displayed.

[Actual results]:
- There is a noticeable delay (depends on how large is the file) until the progress bar is displayed.

[Notes]:
- The user probably think that the download/upload is not working.
- The issue is also reproducible when downloading a large file.
- This issue is also reproducible on the stage page (https://send.stage.mozaws.net). 
- Here is a screen recording with the issue:
delay until progres bar

### LINKED ISSUE #273: UI does not respond when a large file is dropped in
## STR

Drag in a large file and drop it

## Expected
Some kind of immediate indication that your file has been added

## Actual
Hover state persists on drag/drop box, and page feels frozen until redirect to uploading screen

@chenmorpheus, could you guys come up with something here?

### LINKED ISSUE #277: Audit usage of .html()
I noticed in @dannycoates' recent #266 PR that he converted some `.html()` calls to `.text()`. :+1:

We should do a quick double check on some other `$.html()` usages and see if we can tighten them up as well:

```sh
$ git grep -i ".html("
```

> 1. [frontend/src/download.js](https://github.com/mozilla/send/blob/fbf906fe5b1eb501f7d0881fbd4a3f3169525841/frontend/src/download.js#L12):  `const filename = $('#dl-filename').html();`
> 1. [frontend/src/download.js](https://github.com/mozilla/send/blob/fbf906fe5b1eb501f7d0881fbd4a3f3169525841/frontend/src/download.js#L32):      ```$('.percent-number').html(`${Math.floor(percent * 100)}`);```
> 1. [frontend/src/download.js](https://github.com/mozilla/send/blob/fbf906fe5b1eb501f7d0881fbd4a3f3169525841/frontend/src/download.js#L40):                       `$('.title').html(translated[1]);`
> 1. [frontend/src/upload.js](https://github.com/mozilla/send/blob/fbf906fe5b1eb501f7d0881fbd4a3f3169525841/frontend/src/upload.js#L45):    `$copyBtn.html('</img>');`
> 1. [frontend/src/upload.js](https://github.com/mozilla/send/blob/fbf906fe5b1eb501f7d0881fbd4a3f3169525841/frontend/src/upload.js#L115):        ```$('.percent-number').html(`${Math.floor(percent * 100)}`);```
> 1. [frontend/src/upload.js](https://github.com/mozilla/send/blob/fbf906fe5b1eb501f7d0881fbd4a3f3169525841/frontend/src/upload.js#L270):        `$(link).html(linkImg);`
> 1. [frontend/src/upload.js](https://github.com/mozilla/send/blob/fbf906fe5b1eb501f7d0881fbd4a3f3169525841/frontend/src/upload.js#L324-L329):    `$popupText.html([`

### PR DESCRIPTION
**Pull request by dannycoates**
A riff on #282. Thanks @pdehaan!

fixes #128 
fixes #273 
fixes #277

The main difference is the addition of the `setTimeout` to allow the upload-progress panel to show faster.

### CODE REVIEW

**Comment by pdehaan** [comment]
This looks "funny" to my untrained eye:

firefox_send

**Comment by pdehaan** [comment]
Apart from my one nit, looks good to me. Great riff!
Feel free to kill #282.

**Comment by dannycoates** [comment]
> This looks "funny"

yeah. It's less funny with the 100% intact.

File: frontend/src/upload.js [hunk 1]
```diff
@@ -245,76 +243,81 @@ $(document).ready(function() {
...
-        storage.addFile(info.fileId, fileData);
-        $('#upload-filename').attr(
-          'data-l10n-id',
-          'uploadSuccessConfirmHeader'
-        );
-        t = window.setTimeout(() => {
+    // For large files we need to give the ui a tick to breathe and update
+    // before we kick off the FileSender
+    setTimeout(() => {
+      //don't allow drag and drop when not on page-one
```
Comment (ericawright) [inline frontend/src/upload.js]: can you put this comment back with the code it belongs to:
` $('body').off('drop', onUpload);` line 166

**Comment by ericawright** [comment]
I think I'd like it to start with the 0% in the center of the circle even when the other stages are being done.
other than that it's great!

**Comment by dannycoates** [comment]
> I think I'd like it to start with the 0% in the center of the circle even when the other stages are being done.

fixed

File: frontend/src/upload.js [hunk 2]
```diff
@@ -245,76 +240,81 @@ $(document).ready(function() {
...
+    // before we kick off the FileSender
+    setTimeout(() => {
+      fileSender
+        .upload()
+        .then(info => {
+          const endTime = Date.now();
+          const totalTime = endTime - startTime;
+          const uploadTime = endTime - uploadStart;
+          const uploadSpeed = file.size / (uploadTime / 1000);
+          const expiration = 24 * 60 * 60 * 1000; //will eventually come from a field
```
Comment (pdehaan) [inline frontend/src/upload.js]: Unrelated, but not sure if this can be replaced with the `ONE_DAY_IN_MS` const imported above.

**Comment by abhinadduri** [comment]
Looks good!

**Pull request merged**